app/bin/service/search.dart

```diff
@@ -51,9 +51,16 @@ void _main(int isolateId) {
 
     registerSearchBackend(new SearchBackend(db.dbService));
 
-    final Bucket bucket = await _createOrGetBucket(
+    final Bucket snapshotBucket = await _createOrGetBucket(
         storageService, activeConfiguration.searchSnapshotBucketName);
-    registerSnapshotStorage(new SnapshotStorage(storageService, bucket));
+    registerSnapshotStorage(
+        new SnapshotStorage(storageService, snapshotBucket));
+
+    final Bucket popularityBucket = await _createOrGetBucket(
+        storageService, activeConfiguration.popularityDumpBucketName);
+    registerPopularityStorage(
+        new PopularityStorage(storageService, popularityBucket));
+    await popularityStorage.init();
```

Comment:
Be a bit careful when to `await` during initialization, if we wait too long until we call `runAppEngine` then GAE will think the service is unhealthy and kill the VM. Guess it should not take too long to fetch the latest popularity data?

Reply:
I think the fetch and processing shouldn't take longer than a couple of seconds. I think it is safe for now to assume that the storage is available and ready, and later we could add error handling if there is an issue with that. (e.g. the UI could just hide the related data)
However such "deep" error handling is not something I'd like to focus on at the moment, because I think there is very little risk for it, and the current "mock_scores" fallback is good enough for now.